src/rust/container/lib.rs

```diff
@@ -0,0 +1,88 @@
+use bollard::API_DEFAULT_VERSION;
+use bollard::Docker;
+use bollard::exec::CreateExecOptions;
+use bollard::exec::StartExecOptions;
+use bollard::query_parameters::StopContainerOptionsBuilder;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum ContainerError {
+    #[error("Connection error: {0}")]
+    ConnectionError(String),
+    #[error("Container error: {0}")]
+    DestroyError(String),
+    #[error("Container error: {0}")]
+    ExecError(String),
+}
+
+#[cxx::bridge(namespace = "workerd::rust::container")]
+mod ffi {
+    extern "Rust" {
+        pub async fn start(
+            container_name: &str,
+            entrypoint: Vec<&str>,
+            env: Vec<&str>,
+        ) -> Result<String>;
+        pub async fn destroy(container_name: &str) -> Result<()>;
+        pub async fn signal(container_name: &str, signal: &str) -> Result<()>;
+    }
+}
+
+// TODO:
+// [ ] - kj::Promise<void> status(StatusContext context) override;
+// [x] - kj::Promise<void> start(StartContext context) override;
+// [ ] - kj::Promise<void> monitor(MonitorContext context) override;
+// [x] - kj::Promise<void> destroy(DestroyContext context) override;
+// [x] - kj::Promise<void> signal(SignalContext context) override;
+// [ ] - kj::Promise<void> getTcpPort(GetTcpPortContext context) override;
+// [ ] - kj::Promise<void> listenTcp(ListenTcpContext context) override;
+
+pub fn connect() -> Result<Docker, ContainerError> {
+    let addr = "/var/run/docker.sock";
+    let socket = Docker::connect_with_unix(addr, 30, API_DEFAULT_VERSION)
+        .map_err(|e| ContainerError::ConnectionError(e.to_string()))?;
+    Ok(socket)
+}
+
+pub async fn start(
+    container_name: &str,
+    entrypoint: Vec<&str>,
+    env: Vec<&str>,
+) -> Result<String, ContainerError> {
+    // TODO(soon): Persist connection.
+    let socket = connect()?;
```

Comment:
It looks like the lib provides a higher level API [start_container](https://docs.rs/bollard/latest/bollard/struct.Docker.html#method.start_container) ([examples](https://docs.rs/bollard/latest/bollard/#examples))

Could it be used instead?

Reply:
I looked at it once again, and start_exec() isn't the correct command.

Start container requires building an image. How are we going to build an image? cc @mikenomitch